Source/loadsave.cpp

```diff
@@ -638,7 +638,7 @@ void LoadMonster(LoadHelper *file, Monster &monster)
 	monster.whoHit = file->NextLE<int8_t>();
 	monster.level = file->NextLE<int8_t>();
 	file->Skip(1); // Alignment
-	monster.exp = file->NextLE<uint16_t>();
+	monster.exp = file->NextLE<uint32_t>();

```

Comment:
you need to ensure that the same number of bytes are read/written to save files for compatibility with old versions and the base game. You could work around the issue by splitting the value into two locations or writing a truncated value here and a full value somewhere else.

e.g.
```cpp
// To allow for values which overflow the old 16bit space allocated to monster exp we store an extra 8 bits in a previously unused space in the file
monster.exp = file->NextLE<uint8_t>() << 16;
monster.exp |= file->NextLE<uint16_t>();
```

(and do the inverse when saving)

Reply:
since diablo is the only monster that overflows it, I'd suggest just limiting his exp to max possible value, otherwise it's an overkill and who kills hell diablo for xp xD